### PR TITLE
Add TypeConverter with ability to convert from string to ConnectionStringBase

### PR DESCRIPTION
**Pull request by andrewvk**
* Add TypeConverter with ability to convert from string to ConnectionStringBase.
* Remove the IsAssignableTo method from .NET 5+, since a full analogue appeared there.

### CODE REVIEW

File: CodeJam.Main/ConnectionStrings/ConnectionStringBase.ConnectionStringBaseTypeConverter.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+			public ConnectionStringBaseTypeConverter(Type callingType)
+			{
+				Code.AssertArgument(
+					callingType.IsAssignableTo<ConnectionStringBase>(), nameof(callingType),
+					$"To be used with this converter, the type must be inherited from {nameof(ConnectionStringBase)}");
+				_callingType = callingType;
+			}
+
+			private Func<string, object> GetCreator() => _creators.GetOrAdd(
+				_callingType, ct =>
```
Comment (ig-sinicyn) [inline CodeJam.Main/ConnectionStrings/ConnectionStringBase.ConnectionStringBaseTypeConverter.cs:30]: I'm not sure but there may be some helpers for object creation in the CJ. InfoOf / TypeDescriptor if I rememper it right.

If there are nothig suitable, the code looks good for me^)

File: CodeJam.Main/ConnectionStrings/ConnectionStringBase.cs [hunk 2]
```diff
@@ -260,5 +222,45 @@ public bool EquivalentTo(ConnectionStringBase other)
 		/// <inheritdoc />
 		public override string ToString() => _wrapper.ToString();
 		#endregion
+
+		#region KeywordDescriptor class
+		/// <summary>
+		/// Descriptor for connection string keyword
+		/// </summary>
+		[PublicAPI]
+		protected class KeywordDescriptor
```
Comment (ig-sinicyn) [inline CodeJam.Main/ConnectionStrings/ConnectionStringBase.cs:231]: As far as I can remember we usually do place nested classes at the beginning of the file.

There were no explicit guideline for this so I'm curious: should we keep existing style or it's ok to place nested types where it seems to be more natural?

**Review by ig-sinicyn — APPROVED** [review]
LGTM 👍

File: CodeJam.Main/ConnectionStrings/ConnectionStringBase.cs (hunk 2 above)
Comment (andrewvk) [reply CodeJam.Main/ConnectionStrings/ConnectionStringBase.cs:231]: Nested classes at the beginning of the file make it very difficult to read the code, so I would still leave it that way.

File: CodeJam.Main/ConnectionStrings/ConnectionStringBase.ConnectionStringBaseTypeConverter.cs (hunk 1 above)
Comment (andrewvk) [reply CodeJam.Main/ConnectionStrings/ConnectionStringBase.ConnectionStringBaseTypeConverter.cs:30]: If you mean TypeAccessor<T>.Create, unfortunately this method only supports constructors without parameters. You can add overloads Create<TParam1, TParam2, ..>, of course, but that's a separate PR.

**Pull request merged**